gt/main.c

```diff
@@ -131,36 +129,78 @@ gt_parse_incoming_pkt(struct rte_mbuf *pkt, struct gt_packet_headers *info)
 		info->inner_ip_ver = ETHER_TYPE_IPv4;
 		info->l4_proto = inner_ipv4_hdr->next_proto_id;
 		info->l4_hdr = &inner_ipv4_hdr[1];
+
+		if (rte_ipv4_frag_pkt_is_fragmented(inner_ipv4_hdr)) {
```

Comment:
Shouldn't we be performing this fragmentation check on the outer IP header? If one of these packets is fragmented, I think only the first fragment will have the inner IP header.

Reply:
Do you mean the packets are fragmented after encapsulation? I think we are performing this fragmentation on the packets before encapsulation, as stated in the issue https://github.com/AltraMayor/gatekeeper/issues/26 feature 1.

Reply:
I see, I think I misunderstood where these fragmented packets are coming from. This patch is for fragmented packets that arrived at Gatekeeper already fragmented, and therefore if we're sending a packet to Grantor that has been fragmented, if it needs to make a policy decision by reassembling the fragments first.

So yes, all fragmentation will happen before encapsulation, and therefore all fragments will have both the outer and inner header, but the fragmentation is checked against the inner header.